snapcraft/parts/parts.py

```diff
@@ -79,6 +80,7 @@ def __init__(
                 work_dir=work_dir,
                 cache_dir=cache_dir,
                 ignore_local_sources=["*.snap"],
+                project_name=project_name if project_name else "",
```

Comment:
how would project name ever be empty?

Reply:
Oops, good point.  I don't know why I made it optional.